### PR TITLE
Feature/keep value

### PR DESCRIPTION
**Pull request by Waterstraal**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #2 

## What is the new behavior?

There is a new operator called `retainResult` which can save values across emissions.

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: projects/ngneat/loadoff/src/lib/retainResult.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ * This can be useful for implementing a refresh mechanism, where you want to keep displaying the old value while fetching a new value
+ */
+export function retainResult<T, E = any>(): OperatorFunction<AsyncState<T, E>, AsyncState<T, E>> {
+  return pipe(
+    startWith(createAsyncState<T, E>()),
+    scan(
+      (acc: AsyncState<T, E>, val: AsyncState<T, E>) =>
+        new AsyncState<T, E>({
+          ...val,
+          res: val.res ?? acc.res,
```
Comment (NetanelBasal) [inline projects/ngneat/loadoff/src/lib/retainResult.ts:16]: What if the next result is null or undefined and it's considered valid response?

Comment (Waterstraal) [reply projects/ngneat/loadoff/src/lib/retainResult.ts:16]: Good point!

File: README.md [hunk 2]
```diff
@@ -179,6 +179,37 @@ class UsersComponent {
 }
 ```
 
+### `retailResult`
```
Comment (NetanelBasal) [inline README.md:182]:
```suggestion
### `retainResult`
```

File: projects/ngneat/loadoff/src/lib/retainResult.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import { scan, startWith } from 'rxjs/operators';
+import { AsyncState, createAsyncState } from './toAsyncState';
+
+/**
+ * Operator that retains the previous result on a new emission of an AsyncState item.
+ * This can be useful for implementing a refresh mechanism, where you want to keep displaying the old value while fetching a new value
+ */
+export function retainResult<T, E = any>(): OperatorFunction<AsyncState<T, E>, AsyncState<T, E>> {
+  return pipe(
+    startWith(createAsyncState<T, E>()),
```
Comment (NetanelBasal) [inline projects/ngneat/loadoff/src/lib/retainResult.ts:11]: Do you think the startWith should be here or let the consumer define it?

Comment (Waterstraal) [reply projects/ngneat/loadoff/src/lib/retainResult.ts:11]: We could add an optional startWithValue param. Something like the code below, what do you think?

```
export function retainResult<T, E = any>(startWithValue: AsyncState<T, E> = createAsyncState<T, E>()): OperatorFunction<AsyncState<T, E>, AsyncState<T, E>> {
  return pipe(
    startWith(startWithValue),
```

**Pull request merged**